spec/models/reservation_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe Reservation, type: :model do
+  describe 'test reservation validation' do
+    let(:user) { User.create(username: 'JohnDoe') }
+    let(:laptop) do
+      Laptop.create(
+        name: 'DELL Latitude 7280',
+        description: 'Cool',
+        photo_url: 'My photo',
+        price: 440_000,
+        model_year: '2021', rom_size: '500gb',
+        ram_size: '16gb'
+      )
+    end
+
+    before do
+      Reservation.create(user:, laptop:, date: Date.today, city: 'New York', hour: '09:00 AM')
+    end
+
+    it 'should belong to a user' do
+      reservation = Reservation.new(user: nil, laptop:, date: Date.today, city: 'New York', hour: '09:00 AM')
+      expect(reservation).not_to be_valid
+    end
+
+    it 'should belong to a laptop' do
+      reservation = Reservation.new(user:, laptop: nil, date: Date.today, city: 'New York', hour: '09:00 AM')
+      expect(reservation).not_to be_valid
+    end
+
+    it 'should have a date' do
+      reservation = Reservation.new(user:, laptop:, date: nil, city: 'New York', hour: '09:00 AM')
+      expect(reservation).not_to be_valid
+    end
+
+    it 'should have a city' do
+      reservation = Reservation.new(user:, laptop:, date: Date.today, city: nil, hour: '09:00 AM')
+      expect(reservation).not_to be_valid
+    end
+
+    it 'should have an hour' do
+      reservation = Reservation.new(user:, laptop:, date: Date.today, city: 'New York', hour: nil)
+      expect(reservation).not_to be_valid
+    end
+
+    it 'should be unique for a specific laptop on a specific date and hour' do
```

Comment:
It looks like there are some tests it could improve (this is the first one on the list) I kindly invite you to fix this before requesting another code review. 

```
Finished in 0.13423 seconds (files took 1.13 seconds to load)
25 examples, 4 failures

Failed examples:

rspec ./spec/models/reservation_spec.rb:46 # Reservation test reservation validation should be unique for a specific laptop on a specific date and hour
rspec ./spec/requests/api/v1/laptops_spec.rb:37 # laptops /api/v1/laptops post Users created returns a 201 response
rspec ./spec/requests/api/v1/reservation_spec.rb:20 # reservation /api/v1/reservation/ post Reservation created returns a 201 response
rspec ./spec/requests/api/v1/users_spec.rb:35 # laptops /api/v1/login post User created returns a 201 response
```